<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Session1.utf8</title>
    <meta charset="utf-8" />
    <meta name="author" content="Rockefeller University, The Vertebrate Genome Lab" />
    <link href="Session1_files/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="Session1_files/remark-css-0.0.1/default-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# RU Genome Assembly, Session 1
<html>
<div style="float:left">

</div>
<hr color='#EB811B' size=1px width=796px>
</html>
### Rockefeller University, The Vertebrate Genome Lab
### <a href="http://rockefelleruniversity.github.io/RU_course_template/" class="uri">http://rockefelleruniversity.github.io/RU_course_template/</a>

---




# An introduction to genome assembly
#### Principles and practicalities of assembling, evaluating and using a reference genome

&lt;div style="float:left; margin:10px" width="40%"&gt;
Giulio Formenti, Ph.D.&lt;br/&gt;
Laboratory of &lt;br/&gt; Neurogenetics of Language&lt;br/&gt;
Vertebrate Genome Laboratory&lt;br/&gt;
The Rockefeller University&lt;br/&gt;
&lt;a href="mailto:gformenti@rockefeller.edu"&gt;gformenti@rockefeller.edu&lt;/a&gt;
&lt;/div&gt;

&lt;div style="float:left; margin:10px" width="40%"&gt;
Oliver Fedrigo, Ph.D.&lt;br/&gt;
Vertebrate Genome Laboratory&lt;br/&gt;
The Rockefeller University&lt;br/&gt;
&lt;a href="mailto:ofedrigo@rockefeller.edu"&gt;ofedrigo@rockefeller.edu&lt;/a&gt;
&lt;/div&gt;

&lt;br style="clear:left" /&gt;

__In collaboration with the Vertebrate Genomes Laboratory__

_March 22nd &amp; 25th - Monday &amp; Thursday_

&lt;img src="../imgs/VGL_logo.png" width=30%&gt;
&lt;img src="../imgs/RU_logo.png" width=15%&gt;

---
## Materials

- [Introduction to Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/getting-started.html)
- [Introduction to Snakemake](https://snakemake.readthedocs.io/en/stable/)

- [GEP - Genome evaluation pipeline](https://git.imp.fu-berlin.de/cmazzoni/genome_evaluation_snakemake)
- Path on RU HPC to test data: /fakepath/to/data

---
## The Vertebrate Genomes Project

&lt;img src="../imgs/VGP_phases.png" width=35% /&gt;
&lt;img src="../imgs/VGP_map.png" width=55% /&gt;

&lt;center&gt;
&lt;a href="https://vertebrategenomesproject.org/"&gt;VGP website&lt;/a&gt;&lt;br/&gt;
&lt;img src="../imgs/VGP_logo.png" width=30% /&gt;
&lt;/center&gt;

---
## Many international endeavours
&lt;img src="../imgs/B10K_logo.png" width=30% /&gt;
&lt;img src="../imgs/dToL_logo.png" width=30% /&gt;
&lt;img src="../imgs/GIGA_logo.jpg" width=30% /&gt;

&lt;img src="../imgs/EBP_logo.jpg" width=30% /&gt;
&lt;img src="../imgs/ERGA_logo.jpg" width=30% /&gt;
&lt;img src="../imgs/BAT1K_logo.jpg" width=30% /&gt;

&lt;p style="text-align:center"&gt;
&lt;img src="../imgs/BIOGENOMA_logo.png" width=30% /&gt;
&lt;/p&gt;

---
## Why sequencing the DNA

_«A knowledge of sequences could contribute much &lt;br/&gt;
to our understanding of living matter»_

Frederick Sanger, 1980

&lt;img src="../imgs/science.png" width=70% /&gt;

---
## 

&lt;div style="font-size:30px;background-image: url('../imgs/bg.png'); background-repeat:no-repeat; background-size: 796px; text-align:center; height=600px; padding:200px 0 400px 0" width=100%&gt;&lt;b&gt;Genome assembly:&lt;br/&gt;A little bit of history&lt;/b&gt;&lt;/div&gt;

---
## 

&lt;img src="../imgs/gatc_history.png" /&gt;

To learn more: &lt;a href="https://www.sciencedirect.com/science/article/pii/S2001037019303277"&gt;Giani et al., 2020&lt;/a&gt;

---
## 1953: The sequence of insulin

&lt;img src="../imgs/insulin_a.png" width=500px /&gt;
&lt;img src="../imgs/sanger.png" width=197px /&gt;

_Refined partition chromatography:_

The two chains are separated and fragmented, the fragments are individually read and sequences from each fragment overlapped to yield a complete sequence.

To learn more: &lt;a href="https://www.sciencedirect.com/science/article/pii/S2001037019303277"&gt;Giani et al., 2020&lt;/a&gt;

---
## 1953: The sequence of insulin

&lt;img src="../imgs/insulin_b.png" width=500px /&gt;
&lt;img src="../imgs/sanger.png" width=197px /&gt;

_Refined partition chromatography:_

The two chains are separated and fragmented, the fragments are individually read and sequences from each fragment overlapped to yield a complete sequence.

To learn more: &lt;a href="https://www.sciencedirect.com/science/article/pii/S2001037019303277"&gt;Giani et al., 2020&lt;/a&gt;

---
## 1953: The sequence of insulin


&lt;img src="../imgs/insulin_c.png" width=500px /&gt;
&lt;img src="../imgs/sanger.png" width=197px /&gt;


_Refined partition chromatography:_

The two chains are separated and fragmented, the fragments are individually read and sequences from each fragment overlapped to yield a complete sequence.

To learn more: &lt;a href="https://www.sciencedirect.com/science/article/pii/S2001037019303277"&gt;Giani et al., 2020&lt;/a&gt;

---
## 1979: Shotgun sequencing

Rodger Staden, invents of the first DNA sequencing ‘software’.

In 1982, Sanger uses it to assemble the entire 48,502 bp of bacteriophage Lambda genome.

&lt;img src="../imgs/overlap.png" width=500px /&gt;
&lt;img src="../imgs/staden.png" width=197px /&gt;

To learn more: &lt;a href="https://www.sciencedirect.com/science/article/pii/S2001037019303277"&gt;Giani et al., 2020&lt;/a&gt;

---
## Whole Genome Sequencing

&lt;img src="../imgs/timeline_half.png" width=400px /&gt;

---
## De novo genome assembly

The process of determining the sequence of an organism without existing reference.

&lt;img src="../imgs/humanWGA.jpg" width=800px /&gt;

---
## The HGP produced many tech advancements

&lt;img src="../imgs/Moore_law.png" width=600px /&gt;

---
## Whole Genome Sequencing

&lt;img src="../imgs/timeline.png" width=800px /&gt;

---
## Genbank

&lt;img src="../imgs/Genbank1.png" width=800px /&gt;

---
## Genbank

&lt;img src="../imgs/Genbank2.png" width=800px /&gt;

---
## NGS-based WGS
&lt;div style="text-align:center"&gt;
&lt;div style="float:left"&gt;
&lt;img src="../imgs/science_birds1.png" height=250px /&gt; &lt;br/&gt;

&lt;img src="../imgs/science_birds2.png" height=250px /&gt;
&lt;/div&gt;

&lt;img style="float:left" src="../imgs/bird_tree.png" height=500px /&gt;

&lt;br style="clear:left" /&gt;
&lt;/div&gt;

---
## 

&lt;div style="font-size:30px;background-image: url('../imgs/bg.png'); background-repeat:no-repeat; background-size: 796px; text-align:center; height=600px; padding:200px 0 400px 0" width=100%&gt;&lt;b&gt;Genome assembly:&lt;br/&gt;Technologies&lt;/b&gt;&lt;/div&gt;

---
## Repeats are a problem

&lt;img src="../imgs/Genbank2.png" width=800px /&gt;

---
## Third generation sequencing

&lt;div style="float:left; margin:10px"&gt;
&lt;img src="../imgs/NGS.png" width=300px /&gt;&lt;br/&gt;
&lt;b&gt;NGS&lt;/b&gt;&lt;br/&gt;
- Bridge amplification&lt;br/&gt;
- Short reads&lt;br/&gt;
- High throughput&lt;br/&gt;
- High quality&lt;br/&gt;
&lt;/div&gt;
&lt;div style="float:left; margin:10px"&gt;
&lt;img src="../imgs/TGS.png" width=300px /&gt;&lt;br/&gt;
&lt;b&gt;Pacbio (TGS)&lt;/b&gt;&lt;br/&gt;
- Single molecule&lt;br/&gt;
- Long reads&lt;br/&gt;
- Lower throughput&lt;br/&gt;
- Lower quality&lt;br/&gt;
&lt;/div&gt;

&lt;br style="clear:left" /&gt;

&lt;a href="https://www.elsevier.es/es-revista-revista-mexicana-biodiversidad-91-articulo-the-study-biodiversity-in-era-S1870345314730076"&gt;Escalante et al., 2014&lt;/a&gt;

---
## Third generation sequencing: Pacbio

&lt;img src="../imgs/pacbio1.png" width=450px /&gt;
&lt;img src="../imgs/pacbio2.jpg" width=200px /&gt;

---
## Third generation sequencing: Pacbio

&lt;img src="../imgs/pacbio3.png" width=800px /&gt;

---
## Third generation sequencing: Pacbio

&lt;img src="../imgs/pacbio4.png" width=800px /&gt;

---
## Long read matter

High-quality error-free genome assemblies and annotations are necessary as current 1st and 2nd generation genome sequencing approaches generate numerous errors that cause a variety of problems in downstream analyses. &lt;b&gt;Parts of genes are missing, and some are incorrectly assembled, while others are completely missing from the assemblies despite pieces found in the raw sequence reads.&lt;/b&gt; (Vertebrate Genomes Project)

&lt;img src="../imgs/finch.png" width=800px /&gt;

&lt;a href="https://academic.oup.com/gigascience/article/6/10/gix085/4096262"&gt;Korlach et al., 2017&lt;/a&gt;

---
## Third generation sequencing: nanopore

&lt;img src="../imgs/nanopore2.png" width=340px /&gt;
&lt;img src="../imgs/nanopore3.png" width=380px /&gt;

---
## Linked reads

&lt;img src="../imgs/10x.png" width=600px /&gt;
&lt;img src="../imgs/10x_logo.jpg" width=100px /&gt;

* DNA &amp; RNA (single-cell)
* Formerly 10x Genomics, now &lt;a href="https://www.universalsequencing.com/technology"&gt;Universal Sequencing (TELL-Seq)&lt;/a&gt;

---
## Pacbio Hifi

&lt;img src="../imgs/pacbio_logo.png" width=200px /&gt;

&lt;img src="../imgs/ccs.png" width=100% /&gt;

---
## The Telomere-to-Telomere Consortium (T2T)

An open, community-based effort to generate the first complete assembly of a human genome.

&lt;img src="../imgs/logsdon.png" width=100% /&gt;

&lt;a href="https://www.biorxiv.org/content/10.1101/2020.09.08.285395v1"&gt;Logsdon et al., preprint&lt;/a&gt;

---
## The Telomere-to-Telomere Consortium (T2T)

&lt;div style="text-align:center"&gt;
&lt;img src="../imgs/miga1.png" width=75% /&gt;

&lt;img src="../imgs/miga2.png" width=65% /&gt;
&lt;/div&gt;
&lt;a href="https://www.nature.com/articles/s41586-020-2547-7"&gt;Miga et al., 2020&lt;/a&gt;

---
## The Telomere-to-Telomere Consortium (T2T)

Hifi reads are nearly perfect in homopolymer-compressed space.

AATTCTACTCATAT__AAAAA__TCA__TTTTTT__CA &amp;#8594; AATTCTACTCATAT__A__TCA__T__CA

&lt;img style="margin:10px" src="../imgs/NGS.png" width=45% /&gt;
&lt;img style="margin:10px" src="../imgs/TGS.png" width=45% /&gt;

&lt;a href="https://www.elsevier.es/es-revista-revista-mexicana-biodiversidad-91-articulo-the-study-biodiversity-in-era-S1870345314730076"&gt;Escalante et al., 2014&lt;/a&gt;

---
## The Telomere-to-Telomere Consortium (T2T)

&lt;img src="../imgs/nurk.png" width=100% /&gt;

Nurk et al., in preparation

---
## From reads to contigs

&lt;img src="../imgs/reads2contigs.png" width=100% /&gt;

&lt;div style="text-align:center"&gt;&lt;img src="../imgs/swallow.png" width=90% /&gt;&lt;/div&gt;

---
## From tigs to scaffolds

&lt;img src="../imgs/tigs2scaffolds.png" width=100% /&gt;

---
## From scaffolds to chromosomes

&lt;img src="../imgs/scaffolds2chr.png" width=100% /&gt;

---
## Scaffolding: Optical maps

&lt;img style="position:absolute; position: 0 0" src="../imgs/bionano1.png" width=20% /&gt;
&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;
&lt;img src="../imgs/bionano2.jpg" width=100% /&gt;

---
## Scaffolding: Optical maps

&lt;img src="../imgs/bionano3.jpg" width=100% /&gt;
&lt;img src="../imgs/bionano4.png" width=100% /&gt;

---
## Scaffolding: Hic

Long-range interactions. Used also to reconstruct the 3D structure of DNA.

&lt;img src="../imgs/hic1.png" width=45% /&gt;
&lt;img src="../imgs/hic2.png" width=45% /&gt;

---
## High molecular DNA required

&lt;img src="../imgs/PFGE_FEMTO.png" width=100% /&gt;

&lt;a href="https://academic.oup.com/gigascience/article/8/1/giy142/5202456"&gt;Formenti et al., 2019&lt;/a&gt;

---
## VGP pipeline

&lt;img src="../imgs/VGP_pipeline.png" width=100% /&gt;

&lt;a href="https://www.biorxiv.org/content/10.1101/2020.05.22.110833v1"&gt;Rhie et al., in press&lt;/a&gt;

---
## 

&lt;div style="font-size:30px;background-image: url('../imgs/bg.png'); background-repeat:no-repeat; background-size: 796px; text-align:center; height=600px; padding:200px 0 400px 0" width=100%&gt;&lt;b&gt;Genome assembly:&lt;br/&gt;Principles and practicalities&lt;/b&gt;&lt;/div&gt;

---
## Everything starts from an alignment

- Brute-force
- Global alignments
- Local alignments
- Mixed global/local alignments
- Alignment-free approaches

---
## Brute-force

&lt;ol&gt;
&lt;li&gt;Generate all possible alignments&lt;/li&gt;
&lt;li&gt;Calculate quality score for each alignment&lt;/li&gt;
&lt;li&gt;Pick the alignment with the highest quality score&lt;/li&gt;
&lt;li&gt;Done!&lt;/li&gt;
&lt;/ol&gt;

Unfortunately, there are: &lt;img src="../imgs/brute_force.png" width=10% /&gt; potential alignments between 2 sequences of length N

That is, with sequence length = 100:

2^(2\*200)/(3.14\*100)^(1/2) = 9.068476 × 10^58 alignments

---
## Global alignments

&lt;b&gt;1970: Needleman–Wunsch algorithm&lt;/b&gt;

- Still guarantees that the best solution will be found
- Also referred as optimal matching algorithm
- Divides a large problem (that of aligning long sequences between each other) into smaller problems (a concept referred to as Dynamic programming)
- Maximizes similarity
- Goes from the smaller problems to the general problem solution
- Used for ‘optimal’ global alignment
- Generally for highly similar sequences (in sequence and length)
- Very slow as the length of sequences grows, but of utmost precision

---
## 

&lt;div style="float:left"&gt;
&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;
Sequences&lt;br/&gt;
---------&lt;br/&gt;
GCATGCU&lt;br/&gt;
GATTACA&lt;br/&gt;
&lt;br/&gt;
Best alignments&lt;br/&gt;
---------------&lt;br/&gt;
GCATG-CU&lt;br/&gt;
G-ATTACA&lt;br/&gt;
&lt;br/&gt;
GCA-TGCU&lt;br/&gt;
G-ATTACA&lt;br/&gt;
&lt;br/&gt;
GCAT-GCU&lt;br/&gt;
G-ATTACA&lt;br/&gt;
&lt;/div&gt;

&lt;img style="float:left" src="../imgs/NW_algorithm.png" width=70% /&gt;

&lt;br style="clear:left" /&gt;

&lt;a href="https://en.wikipedia.org/wiki/Needleman%E2%80%93Wunsch_algorithm"&gt;Wikipedia&lt;/a&gt;


---
## The edit distance

- Also referred as Levenshtein distance

banana → bamana (substitution of "n" for "m")

bamana → bambna (substitution of "a" for "b")

bambna → bambina (insertion of "i").

EDIT DISTANCE = 3

You can calculate the edit distance for all possible alignments and choose the alignment that minimizes the edit distance (for longer sequences find an heuristic). It has been shown it is mathematically equivalent to optimal matching.

---
##  Local alignments

&lt;b&gt;1981: Smith–Waterman algorithm&lt;/b&gt;

- Compares segments of all possible lengths and optimizes the similarity measure

- The main difference to the Needleman–Wunsch algorithm is that negative scoring matrix cells are set to zero, making positively scoring local alignments visible. Traceback procedure starts at the highest scoring matrix cell and proceeds until a cell with score zero is encountered, yielding the highest scoring local alignment

- It finds the optimal “local” alignment (best local solution)

- For slightly divergent sequences

- Quadratic complexity in time and space, therefore it often cannot be practically applied to large-scale problems → you need linear solutions

---
##  Alignment-free sequence comparisons

Alignment-free approaches have been used for:

- Transcript quantification → 10-100 times faster.
- SNP calling for known variants (genotyping).
- _De novo_ genome assembly.
- Metagenomics.
- Identification of highly divergent sequences.
- Detection of functional similarities between regulatory elements (e.g. promoters, enhancers, and silencers).
- Study of horizontal gene transfer.
- DNA barcoding of species.

---
## Euclidean distance

&lt;div style="text-align:center"&gt;
&lt;img src="../imgs/Euclidean_distance.png" width=70% /&gt;
&lt;br/&gt;
&lt;a href="https://link.springer.com/article/10.1186/s13059-017-1319-7"&gt;Zielezinski et al. 2017&lt;/a&gt;
&lt;/div&gt;

---
## Assembling a genome today

&lt;div style="float:left; margin:20px"&gt;
&lt;b&gt;Short read assemblers:&lt;/b&gt;&lt;br/&gt;
&lt;br/&gt;
- SGA&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;String graph&lt;/i&gt;&lt;br/&gt;
- ValVel&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;String graph&lt;/i&gt;&lt;br/&gt;
- DISCOVAR&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;DBG&lt;/i&gt;&lt;br/&gt;
- SOAPdenovo&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;DBG&lt;/i&gt;&lt;br/&gt;
- Euler&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;DBG&lt;/i&gt;&lt;br/&gt;
- ABySS&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;DBG&lt;/i&gt;&lt;br/&gt;
- Velvet&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;DBG&lt;/i&gt;&lt;br/&gt;
- SPAdes&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;DBG&lt;/i&gt;&lt;br/&gt;
- Edena&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;OLC&lt;/i&gt;&lt;br/&gt;
- Ray&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;Hybrid&lt;/i&gt;&lt;br/&gt;
- SSAKE&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;Greedy&lt;/i&gt;&lt;br/&gt;
- Perga&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;Greedy&lt;/i&gt;&lt;br/&gt;
- ...&lt;br/&gt;
&lt;/div&gt;

&lt;div style="float:left; margin:20px"&gt;
&lt;b&gt;Long read assemblers:&lt;/b&gt;&lt;br/&gt;
&lt;br/&gt;
- Hifiasm&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;OLC&lt;/i&gt;&lt;br/&gt;
- Canu/HiCanu (ex Celera)&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;OLC&lt;/i&gt;&lt;br/&gt;
- Peregrine&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;HGAP/OLC&lt;/i&gt;&lt;br/&gt;
- Falcon-Unzip&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;HGAP/OLC&lt;/i&gt;&lt;br/&gt;
- Flye&amp;nbsp;&amp;nbsp;&amp;nbsp;&lt;i&gt;Repeat graph&lt;/i&gt;&lt;br/&gt;
- ...&lt;br/&gt;
&lt;/div&gt;
&lt;br style="clear:left" /&gt;
&lt;a href="https://en.wikipedia.org/wiki/De_novo_sequence_assemblers"&gt;Wikipedia&lt;/a&gt;

---
## Assembling a genome today

&lt;img src="../imgs/OLC_DBG1.png" width=100% /&gt;

---
## Assembling a genome today

&lt;img src="../imgs/OLC_DBG2.png" width=100% /&gt;

---
## Canu

&lt;div style="text-align:center"&gt;
&lt;img src="../imgs/canu.png" width=50% /&gt;
&lt;/div&gt;

---
## Galaxy

&lt;img src="../imgs/galaxy1.png" width=100% /&gt;

---
## Genome assembly with Galaxy Unicycler

&lt;div style="text-align:center"&gt;
&lt;img src="../imgs/galaxy2.png" width=80% /&gt;
&lt;/div&gt;

---
## 

&lt;div style="font-size:30px;background-image: url('../imgs/bg.png'); background-repeat:no-repeat; background-size: 796px; text-align:center; height=600px; padding:200px 0 400px 0" width=100%&gt;&lt;b&gt;Genome assembly:&lt;br/&gt;Evaluation&lt;/b&gt;&lt;/div&gt;

---
## Synopsis

&lt;ol&gt;
&lt;li&gt;Estimation of genome parameters (Genomescope)&lt;/li&gt;
&lt;li&gt;General assembly metrics (contiguity, quality)&lt;/li&gt;
&lt;li&gt;Assembly completeness (BUSCO)&lt;/li&gt;
&lt;li&gt;Structural accuracy&lt;/li&gt;
&lt;li&gt;Kmer multiplicity (false duplications, missing kmers)&lt;/li&gt;
&lt;li&gt;Switch rates&lt;/li&gt;
&lt;li&gt;WGA (D-Genies, Mummer, Cactus)&lt;/li&gt;
&lt;ol&gt;

---
## 1. Genome parameters

Unrivalled software for assembly-free estimates is Genomescope (v2.0, 2020)

&lt;img src="../imgs/gs1.png" width=49% /&gt;
&lt;img src="../imgs/gs2.png" width=49% /&gt;

---
## What is a genome assembly?

A genome assembly is the entire genomic sequence derived through a _de novo_ (i.e. reference-free) assembly process of the raw sequencing reads and released by the curators of the genome in the database upon publication.

- Until now a single ‘reference’ individual or pooled individuals in a single reference.
- Haploid.
- Sometimes alternative alleles reported as variants.
- Haplotigs were often completely shuffled as it is unfeasible to determine long blocks of haplotype with NGS.
- Sometimes more than one assembly is present.

---
## The primary assembly

The Primary Assembly constitutes what Genbank curators consider the most up-to-date source for the genomic sequence for this reference.

When available, this is the first result being shown.

Picking a random human gene, the Primary Assembly normally refers to the human genome assembly ‘GRCh38’ (Genome Reference Consortium human build n. 38), the latest release of the long list of high-quality assemblies for the human genome generated since 2001.

&lt;img src="../imgs/assembly1.png" width=100% /&gt;

---
## The primary assembly

&lt;img src="../imgs/assembly2.png" width=100% /&gt;

---
## 2. Contiguity

&lt;img src="../imgs/assembly3.png" width=100% /&gt;

---
## 2. Quality (QV)

&lt;div style="text-align:center"&gt;
&lt;img src="../imgs/qv1.png" width=60% /&gt;

&lt;img src="../imgs/qv2.png" width=60% /&gt;
&lt;/div&gt;

---
## 3. BUSCO scores

&lt;div style="text-align:center"&gt;
&lt;img src="../imgs/busco.png" width=80% /&gt;
&lt;/div&gt;

---
## 4. Structural accuracy

&lt;div style="text-align:center"&gt;
&lt;img src="../imgs/hic2.png" width=70% /&gt;
&lt;/div&gt;

---
## 5. Kmer multiplicity

&lt;div style="text-align:center"&gt;
&lt;img src="../imgs/kmer_multiplicity1.png" width=70% /&gt;
&lt;/div&gt;

---
## 5. Kmer multiplicity

&lt;div style="text-align:center"&gt;
&lt;img src="../imgs/kmer_multiplicity2.png" width=70% /&gt;
&lt;/div&gt;

---
## 6. Switch error rate

&lt;div style="text-align:center"&gt;
&lt;img src="../imgs/switch1.png" width=70% /&gt;

&lt;img src="../imgs/switch2.png" width=70% /&gt;
&lt;/div&gt;

---
## 

&lt;div style="font-size:30px;background-image: url('../imgs/bg.png'); background-repeat:no-repeat; background-size: 796px; text-align:center; height=600px; padding:200px 0 400px 0" width=100%&gt;&lt;b&gt;Genome assembly:&lt;br/&gt;Applications&lt;/b&gt;&lt;/div&gt;

---
## Applications

&lt;div style="text-align:center"&gt;
&lt;img src="../imgs/applications_long_reads.png" width=100% /&gt;
&lt;/div&gt;

---
## Contiguity matters

&lt;div style="text-align:center"&gt;
&lt;img src="../imgs/contiguity.png" width=80% /&gt;
&lt;/div&gt;

---
## Contiguity matters


Differences between two humans:

&lt;img src="../imgs/human_SVs.png" width=100% /&gt;

Human-chimp differences (120 Mb overall):

3x10^7 substitutions → 30 Mb (25%)

5x10^6 indels (&lt;80 bp) → 22 Mb (18%)

7x10^4 SVs (&gt;80 bp) → 68 Mb (57%)

---
## Whole genome alignments

&lt;img src="../imgs/chicken_barn_swallow.png" width=100% /&gt;

---
## Whole genome alignments

&lt;div style="text-align:center"&gt;
&lt;img src="../imgs/chicken_zebra_finch.png" width=50% /&gt;
&lt;/div&gt;

---
## Supergenes in the ruff

&lt;img src="../imgs/ruff.png" width=100% /&gt;

&lt;div style="text-align:center"&gt;
&lt;a href="https://www.nature.com/articles/ng.3443"&gt;Küpper et al., 2015&lt;/a&gt;
&lt;br /&gt;&lt;br /&gt;&lt;br /&gt;
&lt;a href="https://www.scilifelab.se/news/a-supergene-underlies-genetic-differences-in-testosteron-levels-and-sexual-behaviour-in-male-ruff/"&gt;Watch this video!&lt;/a&gt;
&lt;/div&gt;
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
